README.md

```diff
@@ -1,4 +1,45 @@
-## Rusty Merkle Tree Implementation
-**TODO**
-- [ ] Write REAMDE
+# Rusty Merkle Tree Implementation
 
+Simple Merkle tree implementation in Rust.
+
+## Installation
+
+- Clone the repository
+```bash
+git clone https://github.com/LucasUTNFRD/Merkle-Tree.git && cd Merkle-Tree
+```
+
+## Features
+
+- Create Merkle trees from any data that implements `AsRef<[u8]>`
+- Generate Merkle proofs for leaf verification
+- Verify Merkle proofs
+- Add new elements to existing trees
+- Generic implementation supporting different data types
```

Comment:
Maybe it's just me, but I think this paragraph is a bit redundant considering that the second item says:
"Create Merkle trees from any data that implements `AsRef<[u8]>`"